src/devices/bus/mononcol/slot.cpp

```diff
@@ -0,0 +1,185 @@
+// license:BSD-3-Clause
+// copyright-holders:Fabio Priuli
+
+#include "emu.h"
+#include "slot.h"
+
+#define VERBOSE 0
+#include "logmacro.h"
+
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(MONONCOL_SOCKET, mononcol_socket_device, "mononcol_socket", "Monon Color Socket")
+DEFINE_DEVICE_TYPE(MONONCOL_CARTSLOT, mononcol_cartslot_device, "mononcol_cartslot", "Monon Color Cartridge Slot")
+
+
+//**************************************************************************
+//  device_mononcol_cart_interface
+//**************************************************************************
+
+device_mononcol_cart_interface::device_mononcol_cart_interface(machine_config const &mconfig, device_t &device) :
+	device_interface(device, "genslot"),
+	m_region(*this, DEVICE_SELF),
+	m_rom(nullptr),
+	m_rom_size(0)
+{
+}
+
+device_mononcol_cart_interface::~device_mononcol_cart_interface()
+{
+}
+
+void device_mononcol_cart_interface::rom_alloc(u32 size, int width, endianness_t endian, char const *tag)
+{
+	if (m_rom)
+	{
+		throw emu_fatalerror(
+				"%s: Request to allocate ROM when already allocated (allocated size %u, requested size %u)\n",
+				device().tag(),
+				m_rom_size,
+				size);
+	}
+
+	std::string fulltag(tag);
+	fulltag.append(MONONCOL_ROM_REGION_TAG);
```

Comment:
You should really use `device().subtag(...)` to compute relative tags rather than relying on string manipulation.

Reply:
you might want to fix this in generic/slot.cpp then, as I'm unlikely to be the only one using it as the most basic example

Reply:
also I have no idea how subtag applies to the operation here, nor is there any bus device using that.  please tell me exactly what you want to change, or this will have to remain as it is.

Reply:
This is the standard pattern, even if the bus devices I've been asked to look at to find a better example eg. nes_slot

Reply:
Pass a relative tag (i.e. without a leading colon) to `subtag` to get a child tag.  Don’t resort to string manipulation unnecessarily.

`generic` isn’t a good example to copy because it’s trying to be everything to everyone so you end up with a pile of completely unnecessary glue for flexibility you don’t need, and `nes_slot` is decidedly *not* a good bus, either (in its current state, it’s a great example of a leaky abstraction, which you decidedly don’t want.  Most of the cartridge slots are still in terrible shape.

Current cartridge slots in fairly good shape are `vboy` which I cleaned up a while ago, and `vcs` which @wilbertpol has just recently overhauled.

Reply:
See https://github.com/mamedev/mame/commit/27d1b900e2715d67ea2633d6b64f2efbb8c5b7f5